scout/info.py

```diff
@@ -46,12 +46,21 @@ def getArp(ip, username, password):
 def getSpeed(ip, username, password):
     '''
     Function that reports speed of Gi0/0 in Mbps.
+    it will also convert speed to Mbps if in Gbps
     '''
     scoutSshClient = scout.ssh.buildSshClient(ip=ip, username=username, password=password)
     stdin, stdout, stderr = scoutSshClient.exec_command("sho int gi0\n")
     sshOut = stdout.read()
     sshBandwidth = sshOut.decode('ascii').strip("\n").split(",")
-    getBandwidth = sshBandwidth[9].strip("Mbps")
+
+    if "Gbps" in sshBandwidth[10]:
+        getBandwidth = sshBandwidth[10].strip("Gbps")
+        getBandwidth += "000"
+    elif "Gbps" in sshBandwidth[9]:
+        getBandwidth = sshBandwidth[9].strip("Gbps")
+        getBandwidth += "000"
+    else:
+        getBandwidth = sshBandwidth[9].strip("Mbps")
```

Comment:
Thank you for your patience & contribution @chrispro-21 :)

This looks good. I thought of something that may be better to do here:

- I think using a regex would be more predictable than splitting the `show int` output and searching through the list (currently done this way). I tried the following out on my end and it seems to work well (similar to what `fetcher()` does):

~~~
>>> scoutSshClient = ssh.buildSshClient(ip="x.x.x.x", username="Cisco", password="Cisco")
>>> stdin, stdout, stderr = scoutSshClient.exec_command("sho int gi0\n")
>>> apBandwidthRegex = re.compile(r'(1|10|100|1000)(Gbps|Mbps)')
>>> sshOut = stdout.read()
>>> decoded = sshOut.decode('ascii').strip("\n")
>>> bandwidth = apBandwidthRegex.search(decoded).group(0)
>>> print(bandwidth)
100Mbps
~~~

Instead of stripping the bandwidth measurement, could you please implement the preceding flow into `getSpeed()`? I think it's more predictable this way and future proofs the implementation of other AP models.

Other than that, I think this looks good! :+1:

Reply:
Sure thing! I am still new to python and regex :disappointed_relieved: